src/main/java/com/indeed/operators/rabbitmq/reconciliation/RabbitMQClusterReconciler.java

```diff
@@ -67,9 +64,10 @@ public void reconcile(final Reconciliation reconciliation) throws InterruptedExc
 
         if (shouldReconcile(resource)) {
             final RabbitMQCluster cluster = clusterFactory.fromCustomResource(resource);
+            final int currentReplicaCount = determineCurrentReplicaCount(resource);
```

Comment:
Really the key to this entire change - we need to save the replica count before reconciling the StatefulSet.